invenio_base/cli.py

```diff
@@ -45,49 +57,71 @@ def list_entrypoints(entry_point):
             # Store entry points.
             if group_name not in found_entry_points:
                 found_entry_points[group_name] = []
-            for ep in entry_points.values():
-                found_entry_points[group_name].append(str(ep))
+            # Convert the entry point to string before appending
+            for _, ep_value in entry_points.items():
+                found_entry_points[group_name].append(str(ep_value))
 
     for ep_group in sorted(found_entry_points.keys()):
         click.secho(f"{ep_group}", fg="green")
         for ep in sorted(found_entry_points[ep_group]):
             click.echo(f"  {ep}")
 
 
-@instance.command("migrate-secret-key")
-@click.option("--old-key", required=True)
-@with_appcontext
-def migrate_secret_key(old_key):
+@instance.command("migrate-secret-key")  # type: ignore[misc]
+@click.option("--old-key", required=True)  # type: ignore[misc]
+@with_appcontext  # type: ignore[misc]
+def migrate_secret_key(old_key: str) -> None:
     """Call entry points exposed for the SECRET_KEY change."""
     if (
         "SECRET_KEY" not in current_app.config
         or current_app.config["SECRET_KEY"] is None
     ):
         raise click.ClickException("SECRET_KEY is not set in the configuration.")
 
-    migrators = []
-    for ep in set(importlib_metadata.entry_points().get("invenio_base.secret_key", [])):
+    migrators: list[SecretKeyMigrator] = []
+    entry_points_data = importlib_metadata.entry_points()
+    # Handle both importlib_metadata v3.x and v4.x+ APIs
+    available_migrators: list[EntryPoint] = []
+
+    if hasattr(entry_points_data, "select"):
+        # Modern API (importlib_metadata >= 3.6)
```

Comment:
for which python version is `importlib_metadata>=3.6` used?

Reply:
read that it works better with python3.12's type system. The code still supports both old and new APIs, but wanted the latest version that's more compatible with python3.12.